pkg/client/client.go

```diff
@@ -132,132 +151,202 @@ func (c *Client) List(ctx context.Context, namespace string, result runtime.Obje
 	if namespace != "" {
 		r = r.NamespaceIfScoped(namespace, c.Namespaced)
 	}
-	err = r.Resource(c.resource).
+	req := r.Resource(c.resource).
 		Prefix(c.prefix...).
-		VersionedParams(&opts, metav1.ParameterCodec).
-		Timeout(timeout).
-		Do(ctx).
-		Into(result)
-	return
+		VersionedParams(&opts.ListOptions, metav1.ParameterCodec).
+		Timeout(timeout)
+
+	for _, modifier := range opts.RequestModifiers {
+		if modifier != nil {
+			modifier(req)
+		}
+	}
+
+	return req.Do(ctx).Into(result)
 }
 
-func (c *Client) Watch(ctx context.Context, namespace string, opts metav1.ListOptions) (watch.Interface, error) {
+// Watch will attempt to start a watch request with the kube-apiserver for resources in the given namespace (if client.Namespaced is set to true).
+// Results will be streamed too the returned watch.Interface.
+func (c *Client) Watch(ctx context.Context, namespace string, opts ListOptions) (watch.Interface, error) {
 	var timeout time.Duration
```

Comment:
Nit: 

I'm not sure if this is different because this is a watch which is long-lived, but other functions call c.setupCtx() on the context, so it might be good to have here too.

Reply:
You are correct we do not want to add a timeout on the context because the watch command is an HTTP stream that is long-lived.